### ISSUE TITLE
Can't place building within "workingRangeTownHallChunks" range after upgrade

### ISSUE THREAD

**Issue by Oraxian** [ldtteam/minecolonies#2341]
### Minecolonies version

version 1.12.2-0.8.7913, SSP

### Expected behavior

Changing "workingRangeTownHallChunks" in minecolonies.cfg should change the distance from colony center that a new building can be placed.

### Actual behaviour

A new building cannot be placed beyond 8 chunks of colony center (default setting,) regardless of value in "workingRangeTownHallChunks."  

### Steps to reproduce the problem

1. Upgrade an existing colony (mine started in version 1.12.2.-0.8.5081 and has been upgraded frequently) to a version of minecolonies after the "workingRangeTownHall" setting was changed to "workingRangeTownHallChunks."  Behavior is not observed in new worlds - those seem to work fine.

2. Change workingRangeTownHallChunks to a different number of chunks, restart Minecraft.

3. Actual build distance remains 8 chunks from colony center (per coordinates found in "center" in colony1.dat,) regardless of setting.  Other configuration settings seem to behave normally (although not exhaustively tested.)

FYI - entire colony is chunk loaded.  Diamond border of colony displays at the expected range in "workingRangeTownHall" when build tool used, and existing buildings built before upgrade continue to work - only no new buildings can be placed.

*****
A big THANKS to the devs for continued work on this Mod.  I can't remember the last time a mod so dramatically (and positively) changed the way I play the game.  Well done!  Although I now have 83 colonists that are very concerned about the unemployment rate due to a lack of build-able land.  Every day, idle youth gather in the town hall and are observed drinking, smoking, discussing the "good life" of the barbarians and planning their next "creeper tipping" outing to the mines.  This will only lead to trouble...  ;)

**Comment by Raycoms** [comment]
Do you have chunkloaders? Is the area close to the spawn? Did you try relogging?
Did you try visiting those chunks?

I made a small saveguard in it to load at least once the chunks correctly even if your colony size was bigger than 8 chunks in each direction.

**Comment by Oraxian** [comment]
I'm a good distance from spawn (Town Hall is at 1486, 874.)

The entire colony, including desired expansion area is chunk loaded, and it was chunk loaded at the time of the upgrade (I've basically had it chunk loaded from the beginning 4 months ago.)  I didn't notice the issue until some time after the "chunk" update, and I think I applied a few updates happened in between.  I relog a few times per day, so that doesn't fix it (at this point at least.)  I generally update MC about once a week, depending on releases.

I found a few other people with the same issue here on the forums:
https://forum.minecolonies.com/printthread.php?tid=217

I attempted the fix recommended by Auriel_1 (and quite a few other things,) with no success.  I know I can restart to fix the issue, but after 4 months, it's hard to just walk away.  My miners have built a labyrinth that make the Mines of Moria look like a McDonald's play place :)  In any case, I know this is alpha, and I am very grateful for all your hard work!

**Comment by Raycoms** [comment]
Is there anything inside the minecolonies/chunkinfo folder?

2018-03-26 0:10 GMT+01:00 Oraxian <notifications@github.com>:

**Comment by Oraxian** [comment]
Not at the moment, although I removed/readded the mod and deleted the minecolonies folder several times in the course of troubleshooting.  Every time I removed it, I did check that folder (as well as the cache folder) and saw it was empty.  I never saw any data in there until I looked at some backups of my save file from the last few months (about 8 of them,) most either don't have a "chunkinfo" folder, or an empty "chunkinfo" folder.  One save, which was done in the middle of a troubleshooting session, had a single file named "chunk-104_31_0."  Another save from a different, new world I created today (that works properly) had a lot of similar named files.  I'm not sure what these are for, or what should be there.

**Comment by Raycoms** [comment]
I will code something tomorrow morning which can help you to fix this, the
folder of the chunkinfo should contain the data which should be written to
those chunks, atm there is no way to "rewrite it" but I might write a
command tomorrow for it.

2018-03-26 1:05 GMT+01:00 Oraxian <notifications@github.com>:

**Comment by Oraxian** [comment]
I really appreciate it :)  That far exceeds my expectations, and hopefully it will help others as well.  Although my wife will be pretty mad.  I haven't lost this much time to a game in many years.

**Comment by Oraxian** [comment]
I just downloaded the new build, and the claim command seems to have fixed my problem.  Thank you very much!

In case someone else comes to this thread looking for help, this is the syntax of the command I used:

/mc colony claim colony: x range: y add: 1
replace x with your colony ID
replace y with the range from center you want to claim, in chunks (I made sure this matched the config file - not sure if that's important, and didn't want to find out!)

I noticed a large claim takes some time to run (I did range 16 and it took a few minutes,) and there is no feedback in the console currently, so be patient.
